corehq/motech/repeaters/expression/repeaters.py

```diff
@@ -80,6 +139,26 @@ def form_class_name(self):
     def payload_doc(self, repeat_record):
         return CommCareCase.objects.get_case(repeat_record.payload_id, repeat_record.domain).to_json()
 
+    def allowed_to_forward(self, payload):
+        allowed = super().allowed_to_forward(payload)
+        if allowed:
+            transactions = CaseTransaction.objects.get_last_n_recent_form_transaction(payload.case_id, 2)
```

Comment:
Is there a reason for number "2" here or we are just considering the lowest possible number?

Thinking more about this, may be we could have a more specific policy.
I am thinking if it might be a genuine use case where one would expect case to be forwarded by repeaters in a sequence and could end up getting surprised by their repeaters not firing as expected since this skips the repeater silently.
I wonder if its better to set the expectation that a case update via expression repeater WILL NOT fire any other repeater.
All case updates have always been raised via HQ until this point, so may be it's okay to have such a policy.
In this case, we would not fire the repeater here if the last update has XMLNS of repeater response.
Just wanted to put that thought out for consideration.

Reply:
I need it to fire another repeater. Use case:

* Form creates a case and the form triggers a repeater to register the case with an external system. The case is updated with the external system's ID
* Case repeater then forwards the case to a 2nd external system after the external system ID has been set.

To fully support chaining we'd need to load ALL the case transactions and test to see whether there has been a 'non-repeater' case update since the last time THIS repeater fired.

For now I'm happy limiting this to two since I think it's a rare enough use case.

Reply:
Eating my words: https://github.com/dimagi/commcare-hq/pull/35092